core/common/src/main/java/alluxio/Client.java

```diff
@@ -32,8 +34,11 @@
   void disconnect();
 
   /**
-   * @return the {@link InetSocketAddress} of the remote
+   * @return the {@link InetSocketAddress} of the remote, or null if the remote address cannot be
+   *         determined, e.g. when trying to connect to the primary master when leader election is
+   *         in progress
```

Comment:
Consider making this method not return null and require implementing classes to handle the retry logic within getAddress and throw an exception if we exceed retries.